akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -55,6 +58,11 @@
      */
     ResponseEntity entity();
 
+    /**
+     *  Drains entity stream of this message
+     */
+    CompletionStage<Done> ignoreDataBytesJ(Materializer materializer);
```

Comment:
> ```
> // scala
> def method()(implicit mat: Materializer): DiscardedEntity 
> == 
> // java / bytecode
> public DiscardedEntity method(mat: Materializer)
> ```

Scalac detects name clash here as both have the same signature after erasure and both are in the same inheritance scope.

What about mixing two approaches and implementing scala method that returns DiscardedEntity within an implicit class?